modules/core/src/main/java/org/apache/ignite/IgniteSet.java

```diff
@@ -127,4 +127,12 @@
      * @throws IgniteException If job failed.
      */
     public <R> R affinityCall(IgniteCallable<R> job) throws IgniteException;
+
+    /**
+     * Returns set that will operate with binary objects. This is similar to {@link IgniteCache#withKeepBinary()} but
+     * for sets.
+     *
+     * @return New set instance for binary objects.
+     */
+    public <V1> IgniteSet<V1> withKeepBinary();
```

Comment:
I'll leave my main comment here. I see no specific test that checks that binary mode for set actually gives you binary values. Can you please add it somewhere?